### PR TITLE
Add workflow to create infra-deployments PRs

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-1773

### CODE REVIEW

**Comment by lcarva** [comment]
Requires:

- https://github.com/hacbs-contract/ec-cli/pull/413
- https://github.com/hacbs-contract/ec-policies/pull/330
- https://github.com/hacbs-contract/enterprise-contract-controller/pull/66

**Comment by lcarva** [comment]
Requires:

- https://github.com/hacbs-contract/ec-cli/pull/413
- https://github.com/hacbs-contract/ec-policies/pull/330
- https://github.com/hacbs-contract/enterprise-contract-controller/pull/66

**Comment by lcarva** [comment]
I just realized that the secrets are set at the organization level. So no action is needed here. Marked those items as complete.

File: .github/workflows/create-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      with:
+        path: infra-deployments-ci
+
+    - name: Create PR in infra-deployments
+      env:
+        EC_AUTOMATION_KEY: ${{ secrets.EC_AUTOMATION_KEY }}
+        DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+        # TODO: Is this right? Other repos have it as 32872589 but I can't figure out where
+        # that came from.
+        APP_INSTALL_ID: 29980719
```
Comment (zregvart) [inline .github/workflows/create-pr.yaml:69]:
```suggestion
        APP_INSTALL_ID: 32872589
```

This ends up being used to generate `GITHUB_TOKEN` needed for `gh pr create` against https://github.com/redhat-appstudio/infra-deployments/, the installation can be [seen here](https://github.com/apps/ec-automation/installations/32872589) (don't update any permissions there that'll trigger approvals in https://github.com/redhat-appstudio/)

File: .github/workflows/create-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        function createJWT() {
+          local header=$(echo -n '{"alg":"RS256","typ":"JWT"}' | base64 | sed s/\+/-/ | sed -E s/=+$//)
+          local now_utc=$(date --utc +%s)
+          local payload=$(echo -n '{"iat":'$((now_utc - 60))',"exp":'$((now_utc + 120))',"iss":245286}' | base64 | sed s/\+/-/ | sed -E s/=+$//)
+          local signature=$(echo -n "${header}.${payload}" | openssl dgst -sha256 -binary -sign <(echo "${EC_AUTOMATION_KEY}")| base64 | tr -d '\n=' | tr -- '+/' '-_')
+          echo "${header}.${payload}.${signature}"
+        }
+
+        ENSURE_UNIQUE=1 \
```
Comment (zregvart) [inline .github/workflows/create-pr.yaml:83]: Used?
```suggestion
```

**Review by zregvart — APPROVED** [review]

File: .github/workflows/create-pr.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+name: create-pr
+
+on:
+  workflow_dispatch:
+  # TODO: schedule
```
[the remark this answers is not in the document]
Comment (lcarva) [reply .github/workflows/create-pr.yaml:6]: I think we should leave it with just `workflow_dispatch` for the first merge. I'll make a follow up PR to enable a schedule. Once a week sounds like a good starting point.

File: .github/workflows/create-pr.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      with:
+        repository: redhat-appstudio/infra-deployments
+        ref: main
+        path: infra-deployments
+
+    - name: Checkout ec-cli
+      uses: actions/checkout@v3
+      with:
+        repository: hacbs-contract/ec-cli/
+        ref: batching
```
[the remark this answers is not in the document]
Comment (lcarva) [reply .github/workflows/create-pr.yaml]: ... thanks for catching this. sloppy on my part.

**Comment by lcarva** [comment]
All requirements met. Going to merge it so I can try it out.

**Pull request merged**